### PR TITLE
refactor: REST - create rest/builder.nim and simplify factory/app.nim

### PR DESCRIPTION
**Pull request by Ivansete-status**
# Description
Extract some REST logic from app.nim, aiming for simplify `app.nim` a little.
I will submit a couple of similar PRs simplifying `app.nim`.

## Issue
With that simplification, this PR indirectly contributes to the following: migrate DiscV5 and DNS Discovery from app.nim to waku_node.nim https://github.com/waku-org/nwaku/issues/2452

### CODE REVIEW

**Review by SionoiS — APPROVED** [review]
LGTM

I like this a lot!

**Review by NagyZoltanPeter — CHANGES_REQUESTED** [review]
I think we are crossing our works. Can you please take a look at mine. I will draft it now. I'm just adding a final json serialization for a bit extended /health endpoint.
To enable separation of initialization health from other endpoints and start rest server earlier, I needed to rework to order of rest initialization. 
Still, I agree to simplify waku node initialization order due I found its overly complicated.

File: waku/factory/app.nim [hunk 1]
```diff
@@ -456,18 +314,15 @@ proc setupMonitoringAndExternalInterfaces*(app: var App): AppResult[void] =
...
 
   ok()
 
 # App shutdown
 
 proc stop*(app: App): Future[void] {.async: (raises: [Exception]).} =
-  if app.restServer.isSome():
-    await app.restServer.get().stop()
+  if app.conf.rest:
+    await app.restServer.stop()
```
Comment (gabrielmer) [inline waku/factory/app.nim:325]: Not sure I don't understand this change. Shouldn't we check for the `restServer` to be already mounted and not just in the configurations?

**Review by gabrielmer — COMMENTED** [review]
Looks amazing! Thanks so much! Love this change 😍 

Just added a tiny question

File: waku/factory/app.nim (hunk 1 above)
Comment (Ivansete-status) [reply waku/factory/app.nim:325]: Yes makes sense! I'll double-check

**Review by gabrielmer — APPROVED** [review]
Thanks so much!

**Comment by Ivansete-status** [comment]
Closing because has been solved in https://github.com/waku-org/nwaku/pull/2623

**Pull request closed without merging**